sklearn/utils/tests/test_sparsefuncs.py

```diff
@@ -547,3 +548,16 @@ def test_inplace_normalize():
                 if inplace_csr_row_normalize is inplace_csr_row_normalize_l2:
                     X_csr.data **= 2
                 assert_array_almost_equal(np.abs(X_csr).sum(axis=1), ones)
+
+
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_csr_row_norms(dtype):
+    # checks that csr_row_norms returns the same output as
+    # scipy.sparse.linalg.norm, and that the dype is the same as X.dtype.
+    X = sp.random(100, 10, format='csr', dtype=dtype)
+
+    scipy_norms = sp.linalg.norm(X, axis=1)**2
+    norms = csr_row_norms(X)
+
+    assert norms.dtype.type is dtype
```

Comment:
A nit, but I remember having some issues in the past using `is` when checking dtypes: https://stackoverflow.com/questions/26921836/correct-way-to-test-for-numpy-dtype

Reply:
good to know. Changed.